### PR TITLE
Omit non-essential fields from RTAMO flow (Fixes #14482)

### LINKED ISSUE #14482: Remove non-essential attribution fields from RTAMO flow
## Description

As part of the data privacy / consent epic (see https://github.com/mozilla/bedrock/issues/14213), we want to remove non-necessary attribution fields (such as GA client_id and session_id) from the RTAMO flow. Only required fields that are necessary for the feature to function should be passed:

- `utm_source`
- `utm_medium`
- `utm_campaign`
- `utm_content`
- `referrer`

The following should not be set

- `ua`
- `experiment`
- `variation`
- `client_id`
- `client_id_ga4`
- `session_id`

---

#### Success Criteria
- [x] RTAMO only includes required attribution fields

### PR DESCRIPTION
**Pull request by alexgibson**
## One-line summary

This PR removes non-essential attribution fields (see linked issue) from the RTAMO user flow.

## Issue / Bugzilla link

#14482

## Testing

Use a browser with DNT disabled to test:

1. Open https://www-demo3.allizom.org/en-US/firefox/download/thanks/?s=direct&utm_source=test&utm_campaign=test
2. Once the file download has started, open Developer Tools and select the Application tab.
3. In the Storage panel on the left, open the Cookie menu and select www-demo3.
4. Find the cookie called `moz-stub-attribution-code` and copy it's value to the clipboard.
5. Using the command line, decode the attribution string using:

```
echo -n 'paste-attribution-string-here' | base64 --decode
```

You should see in the decoded string that the non-essential fields listed in the issue are `(not set)`.

`source=test&medium=(direct)&campaign=test&content=(not set)&experiment=(not set)&variation=(not set)&ua=other&client_id=(not set)&client_id_ga4=(not set)&session_id=(not set)&dlsource=mozorg
`

 In particular `client_id`, `client_id_ga4` and `session_id` should all be `(not set)`. Note that for `ua` a default value of `other` is expected, instead of the regular browser name.

You can also try https://www.base64decode.org/ for decoding in the browser.

### CODE REVIEW

File: docs/download-buttons.rst [hunk 1]
```diff
@@ -38,40 +38,3 @@ Documentation
 See `helpers.py`_ for documentation and supported parameters for both buttons.
 
 .. _helpers.py: https://github.com/mozilla/bedrock/blob/main/bedrock/firefox/templatetags/helpers.py
-
-External referrers
```
Comment (alexgibson) [inline docs/download-buttons.rst:42]: Note for reviewer: I'm removing this for now as we no longer want to encourage sites from using this, at least without some prior conversation around how they plan to handle data consent on their end. For AMO, legal have approved use here.

File: media/js/base/stub-attribution.js [hunk 2]
```diff
@@ -466,19 +466,31 @@ if (typeof window.Mozilla === 'undefined') {
...
+            : params.get('experiment') || StubAttribution.experimentName;
+        var variation = omitNonEssentialFields
+            ? null
+            : params.get('variation') || StubAttribution.experimentVariation;
         var referrer = StubAttribution.getReferrer(ref);
-        var ua = StubAttribution.getUserAgent();
-        var clientIDUA = StubAttribution.getUAClientID();
-        var clientIDGA4 = StubAttribution.getGtagClientID();
+        var ua = omitNonEssentialFields
+            ? 'other'
```
Comment (willdurand) [inline media/js/base/stub-attribution.js:486]: shouldn't we nullify the `ua` too (for consistency)?

Comment (alexgibson) [reply media/js/base/stub-attribution.js:486]: We could do, but we default to `other` in the case that we can't determine a user agent already. So I figured maybe that makes more sense?

Comment (willdurand) [reply media/js/base/stub-attribution.js:486]: Oh so we never use `(notset)` for `ua`?

**Review by stephaniehobson — APPROVED** [review]
Everything works as expected. Tidy thoughtful PR. R+

Leaving off merging in case https://github.com/mozilla/bedrock/issues/14482#issuecomment-2087933815 is actionable.

**Pull request merged**